cmd/minikube/cmd/config/config_test.go

```diff
@@ -41,12 +41,14 @@ var configTestCases = []configTestCase{
     "ReminderWaitPeriodInHours": 99,
     "cpus": 4,
     "disk-size": "20g",
+    "iso-url": "http://foo.bar/minishift-centos.iso",
     "log_dir": "/etc/hosts",
     "show-libmachine-logs": true,
     "v": 5,
     "vm-driver": "kvm"
 }`,
 		config: map[string]interface{}{
+			"iso-url": "http://foo.bar/minishift-centos.iso",
```

Comment:
> What do you mean other data, can you provide some pointer?

Right now we don't have anything, but for some of the things I have in mind, we need to store some additional state information. I'd like to now whether we can add this to _config.json_ or whether we need a different config file.

> Set method already have test which tell if we put a unknown property and it does throw an error.

Sure, but what if the user edits the file or we add other custom data. Is this code smart enough to deal with this. Also there is a disconnect here. 'minishift set' will make sure to persist the properties listed here. However, Viper does not care about this and reads the plain config file. So if I manually add another flag (which is not listed here) to the config, I think it will be picked up fine by Viper. I think there might be some things we need to think through here.

> If a user manually edit then minishift doesn't care about extra flags or if by mistake user introduced a typo in config that also ignored by minishift.

See this is what I mean. So this code seems to be able to just filter out the settings for _MinikubeConfig_, ignoring other values. This means we could also store other data in _config.json_. What's about config writes? Say I have additional data and read the config, modify it and then save it again (using minishift get/set), does the additional data persist? This are the things I would like to see tests around to define the expected behavior. 

> I think we can add a comment on top of that file and warn user not to edit it manually and use config set command.

-1, I think it is a valid use case to edit the json file by hand.